src/Sylius/Behat/Context/Ui/Admin/ManagingProductAttributesContext.php

```diff
@@ -0,0 +1,309 @@
+    }
+
+    /**
+     * @Then the code field should be disabled
+     */
+    public function theCodeFieldShouldBeDisabled()
+    {
+        Assert::true(
+            $this->updatePage->isCodeDisabled(),
+            'Code field should be disabled, but it does not.'
+        );
+    }
+
+    /**
+     * @Then the type field should be disabled
+     */
+    public function theTypeFieldShouldBeDisabled()
+    {
+       $currentPage = $this->currentPageResolver->getCurrentPageWithForm($this->createPage, $this->updatePage);
+
+        Assert::true(
+            $currentPage->isTypeDisabled(),
+            'Type field should be disabled, but it does not.'
+        );
+    }
+
+    /**
+     * @Then I should be notified that product attribute with this code already exists
+     */
+    public function iShouldBeNotifiedThatProductAttributeWithThisCodeAlreadyExists()
+    {
+        Assert::true(
+            $this->updatePage->checkValidationMessageFor('code', 'This code is already in use.'),
+            'Unique code violation message should appear on page, but it does not.'
+        );
+    }
+
+    /**
+     * @Given there should still be only one product attribute with code :code
+     */
+    public function thereShouldStillBeOnlyOneProductAttributeWithCode($code)
+    {
+        $this->indexPage->open();
+
+        Assert::true(
+            $this->indexPage->isResourceOnPage(['code' => $code]),
+            sprintf('There should be only one product attribute with code %s, but it does not.', $code)
+        );
+    }
+
+    /**
+     * @When I do not name it
+     */
+    public function iDoNotNameIt()
+    {
+        // Intentionally left blank to fulfill context expectation
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatIsRequired($element)
+    {
+        $this->assertFieldValidationMessage($element, sprintf('Please enter attribute %s.', $element));
+    }
+
+    /**
+     * @Given the attribute with :elementName :elementValue should not appear in the store
+     */
+    public function theAttributeWithCodeShouldNotAppearInTheStore($elementName, $elementValue)
+    {
+        $this->indexPage->open();
+
+        Assert::false(
+            $this->indexPage->isResourceOnPage([$elementName => $elementValue]),
+            sprintf('There should not be product attribute with %s %s, but it is.', $elementName, $elementValue)
+        );
+    }
+
+    /**
+     * @When I remove its name from :language translation
+     */
+    public function iRemoveItsNameFromTranslation($language)
+    {
+        $this->updatePage->changeName('', $language);
+    }
+
+    /**
+     * @param string $element
+     * @param string $expectedMessage
+     */
+    private function assertFieldValidationMessage($element, $expectedMessage)
+    {
+        Assert::true(
+            $this->createPage->checkValidationMessageFor($element, $expectedMessage),
```

Comment:
This method should use current page resolver. For example you have checked validation message for name field on update page, but haven't declared name element there.